src/Travis.jl

```diff
@@ -6,93 +6,143 @@ $(EXPORTS)
+        println("\n", read("$filename.pub", String))
+        rm("$filename.pub")
     end
 
-    cd(path) do
-        # Check for old '$filename.enc' and terminate.
-        isfile("$filename.enc") &&
-            error("$package already has an ssh key. Remove it and try again.")
-
-        # Are we in a git repo?
-        success(`git status`) || error("'Travis.genkey' only works with git repositories.")
-
-        # Find the GitHub repo org and name.
-        user, repo =
-            let r = readchomp(`git config --get remote.$remote.url`)
-                m = match(GITHUB_REGEX, r)
-                m === nothing && error("no remote repo named '$remote' found.")
-                m[2], m[3]
-            end
-
-        # Generate the ssh key pair.
-        success(`ssh-keygen -N "" -f $filename`) || error("failed to generated ssh key pair.")
-
-        # Prompt user to add public key to github then remove the public key.
-        let url = "https://github.com/$user/$repo/settings/keys"
-            @info("add the public key below to $url with read/write access:")
-            println("\n", read("$filename.pub", String))
-            rm("$filename.pub")
-        end
+    # Base64 encode the private key and prompt user to add it to travis. The key is
+    # *not* encoded for the sake of security, but instead to make it easier to
+    # copy/paste it over to travis without having to worry about whitespace.
+    let url = "https://travis-ci.org/$user/$repo/settings"
+        @info("add a secure environment variable named 'DOCUMENTER_KEY' to $url with value:")
+        println("\n", base64encode(read(filename, String)), "\n")
+        rm(filename)
+    end
+end
+
+"""
+    genkeys(package::Module; remote="origin")
+
+Like the other method, this generates the SSH keys necessary for the automatic deployment of
+documentation with Documenter from Travis to GitHub Pages, but attempts to guess the package
+URLs from the Git remote.
+
+`package` needs to be the top level module of the package. The `remote` keyword argument can
+be used to specify which Git remote is used for guessing the repository's GitHub URL.
 
-        # Base64 encode the private key and prompt user to add it to travis. The key is
-        # *not* encoded for the sake of security, but instead to make it easier to
-        # copy/paste it over to travis without having to worry about whitespace.
-        let url = "https://travis-ci.org/$user/$repo/settings"
-            @info("add a secure environment variable named 'DOCUMENTER_KEY' to $url with value:")
-            println("\n", base64encode(read(".documenter", String)), "\n")
-            rm(filename)
+This method requires the following command lines programs to be installed:
+
+- `which`
+- `git`
+- `ssh-keygen`
+
+!!! note
+    The package must be in development mode. Make sure you run
+    `pkg> develop pkg` from the Pkg REPL, or `Pkg.develop(\"pkg\")`
+    before generating docs.
+
+# Examples
+
```

Comment:
remove empty line?

Reply:
I usually keep the headings separated with empty lines. I find it helps with readability.